### PR TITLE
feat: manage env code split

### PR DESCRIPTION
**Pull request by ayushpahwa**
Coed split PR for custom environments [EE PR](https://github.com/appsmithorg/appsmith-ee/pull/2207)

### CODE REVIEW

File: app/client/src/ee/pages/Applications/WorkspaceMenu.tsx [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export * from "ce/pages/Applications/WorkspaceMenu";
 import { default as CE_WorkspaceMenu } from "ce/pages/Applications/WorkspaceMenu";
 export default CE_WorkspaceMenu;
+export const EEWorkspaceMenuItems = ({}: { workspaceId: string }) => {
```
Comment (ankitakinger) [inline app/client/src/ee/pages/Applications/WorkspaceMenu.tsx:4]: why is this required?

File: app/client/src/pages/workspace/settings.tsx [hunk 2]
```diff
@@ -230,39 +163,12 @@ export default function Settings() {
...
-            {tabArr.map((tab) => {
-              return (
-                <TabPanel className="tab-panel" key={tab.key} value={tab.key}>
-                  {tab.panelComponent}
-                </TabPanel>
-              );
-            })}
-          </Tabs>
-        </TabsWrapper>
+        <WorkspaceSettingsTabs
```
Comment (ankitakinger) [inline app/client/src/pages/workspace/settings.tsx:166]: Could you handle the 3rd tab scenario in `shouldRedirect` inside this component?

Also, could we please test this out with GAC on EE? Considering when members tab should be visible, when none of the tabs have access so it should redirect? Also, if you manually type the URL you don't have access to, then that should redirect to homepage.

File: app/client/src/ee/pages/Applications/ManageEnvironmentsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const ManageEnvironmentsMenu = ({}: { workspaceId: string }) => {
```
Comment (ankitakinger) [inline app/client/src/ee/pages/Applications/ManageEnvironmentsMenu.tsx:1]: can this be split and added to `ce` folder with null?

File: app/client/src/ce/pages/workspace/Members.tsx [hunk 4]
```diff
@@ -52,6 +52,8 @@ export const MembersWrapper = styled.div<{
   isMobile?: boolean;
 }>`
   &.members-wrapper {
+    overflow: scroll;
+    height: 100%;
```
Comment (ankitakinger) [inline app/client/src/ce/pages/workspace/Members.tsx:56]: why exactly did we remove this for all tabs which was by default done earlier and added it separately for each tab?

Comment (ankitakinger) [reply app/client/src/ce/pages/workspace/Members.tsx:56]: I feel this was better handled earlier because now each tab including the new one you added needs to handle this scrolling separately.

**Review by ankitakinger — CHANGES_REQUESTED** [review]
Added a comment on Members component. Please look at it.

File: app/client/src/pages/workspace/settings.tsx [hunk 5]
```diff
@@ -125,58 +88,31 @@ export default function Settings() {
...
-      location={location}
-      path={`${path}/general`}
-    />
-  );
+  const pageMenuItems: any[] = [
+    {
+      icon: "book-line",
+      className: "documentation-page-menu-item",
+      onSelect: () => {},
+      text: "Documentation",
```
Comment (AmanAgarwal041) [inline app/client/src/pages/workspace/settings.tsx:111]: Can we use `DOCUMENTATION` from message.ts file for text value here ?

**Comment by ayushpahwa** [comment]
Testing completed at EE PR. https://github.com/appsmithorg/appsmith-ee/pull/2209#issuecomment-1794541293

**Review by ankitakinger — APPROVED** [review]
